lib/game/game_server.ex

```diff
@@ -412,6 +413,36 @@ defmodule ThistleTea.Game do
     {:noreply, {socket, state}, socket.read_timeout}
   end
 
+  def handle_cast({:modify, :speed, rate}, {socket, state}) do
+    new_rate = rate * 7.0
+
+    character =
+      state.character
+      |> Map.put(
+        :movement,
+        state.character.movement
+        |> Map.merge(%{run_speed: new_rate})
+      )
+
+    state =
+      Map.merge(state, %{
+        character: character
+      })
+
+    move_event = 0
+
+    run_speed_change_packet =
+      ThistleTea.Util.pack_guid(state.guid) <>
+        <<
+          move_event::little-float-size(32),
+          new_rate::little-float-size(32)
+        >>
+
+    send_packet(@smsg_force_run_speed_change, run_speed_change_packet)
```

Comment:
Didn't know about this packet. If we wanted to be more generic to handle more .modify commands, do you think a `SMSG_UPDATE_OBJECT` would work?

Reply:
Or if we want to be consistent with Mangos, they do use `SMSG_FORCE_RUN_SPEED_CHANGE` for this:

```
2025-01-16 17:44:00 
SERVER:
SOCKET: 23
LENGTH: 10
OPCODE: SMSG_FORCE_RUN_SPEED_CHANGE (0x00E2)
DATA:
01 01 00 00 00 00 00 00 A8 41 
```